paperqa/contrib/zotero.py

```diff
@@ -184,7 +175,14 @@ def iterate(
         while num_remaining > 0:
             cur_limit = min(max_limit, num_remaining)
             self.logger.info(f"Downloading new batch of up to {cur_limit} papers.")
-            _items = self.top(**query_kwargs, limit=cur_limit, start=i)
+
+            if collection_name: 
+                collection_id = self._get_collection_id(collection_name)
+                if not collection_id: continue
+                _items = self.collection_items(collection_id, limit=cur_limit, start=i)
```

Comment:
This line is invalid, see the definition of `collection_items`:

```
    @retrieve
    def collection_items(self, collection, **kwargs):
        """Get a specific collection's items"""
        query_string = "/{t}/{u}/collections/{c}/items".format(
            u=self.library_id, t=self.library_type, c=collection.upper()
        )
        return self._build_query(query_string)
```

so this would also result in an infinite loop

Reply:
Is this comment actually resolved? `start` and `limit` are not actually used here, and if the collection has fewer items than `limit`, I think it will just repeatedly call the collection.

Reply:
Okay I added a suggestion that I think solves it